### PR TITLE
Fix/v1 ext extension

### PR DESCRIPTION
**Pull request by deichmab-draeger**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the [changelog](../CHANGELOG.md) accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/sdc11073/mdib/containerproperties.py [hunk 1]
```diff
@@ -456,45 +456,103 @@ def updateXMLValue(self, instance, node):
...
+    #     self.value = value or list()
+
+    def __eq__(self, other):
+        try:
+            if len(self) != len(other):
+                return False
+        except TypeError:  # len of other cannot be determined
+            return False
+        for my_element, other_element in zip(self, other):
+            if not ExtensionLocalValue.compare_method(my_element, other_element):
```
Comment (leon1995) [inline src/sdc11073/mdib/containerproperties.py:497]: `return all(self.__class__.compare_method(left, right) for left, right in zip(self, other))`

File: src/sdc11073/mdib/containerproperties.py [hunk 2]
```diff
@@ -456,45 +456,103 @@ def updateXMLValue(self, instance, node):
...
-
-            del subNode[:]  # delete all children first
-            if property_value.py_value is not None:
-                subNode.extend([copy.copy(n) for n in property_value.py_value])
+        except AttributeError:
+            return  # nothing to add
+        if property_value is None or len(property_value.py_value) == 0:
+            return  # nothing to add
+        sub_node = self._getElementbyChildNamesList(node, self._subElementNames, createMissingNodes=True)
+        sub_node.extend(copy.copy(x) for x in property_value.py_value)
```
Comment (leon1995) [inline src/sdc11073/mdib/containerproperties.py:555]: use custom copy method

File: src/sdc11073/mdib/descriptorcontainers.py [hunk 3]
```diff
@@ -74,31 +74,19 @@ def codeId(self):
...
-            # delete current retrievability info if it exists
-            for elem in self.ext_Extension:
-                if elem.tag == retrievability_tag:
-                    self.ext_Extension.remove(elem)
-                    break
-        node = retrievability_instance.asEtreeNode(retrievability_tag, nsmap=None)
-        self.ext_Extension.append(node)
+    def get_retrievability(self):
+        """Return all retrievability data from Extension."""
+        return  [msgtypes.Retrievability.fromNode(x) for x in self.ext_Extension
```
Comment (leon1995) [inline src/sdc11073/mdib/descriptorcontainers.py:79]: double whitespace

File: src/sdc11073/mdib/descriptorcontainers.py [hunk 4]
```diff
@@ -74,31 +74,19 @@ def codeId(self):
...
+                 if x.tag == msgTag('Retrievability')]
+
+    def set_retrievability(self, retrievabilities):
+        """Replace all retrievability elements with provided ones in Extension.
+
+        :param retrievabilities: Iterable of msgtypes.Retrievability."""
+        tmp = [x for x in self.ext_Extension if x.tag == msgTag('Retrievability')]
+        for x in tmp:
```
Comment (leon1995) [inline src/sdc11073/mdib/descriptorcontainers.py:87]: no need for temporary variable

File: tests/test_pmtypes.py [hunk 5]
```diff
@@ -31,3 +33,231 @@ def test_base_demographics(self):
...
+                    <ext:Extension xmlns:ext="__ExtensionPoint__">
+                        <foo someattr="somevalue"/>
+                        <bar anotherattr="differentvalue2"/>
+                    </ext:Extension>
+                </pm:Identification>
+                """
+        inst2 = pmtypes.InstanceIdentifier.fromNode(fromstring(another_xml))
+        self.assertNotEqual(inst1.ext_Extension, inst2.ext_Extension)
+        self.assertNotEqual(inst1, inst2)
+
```
Comment (leon1995) [inline tests/test_pmtypes.py:71]: add test `test_compare_extension_with_other_types`

File: tests/test_pmtypes.py [hunk 6]
```diff
@@ -31,3 +33,231 @@ def test_base_demographics(self):
...
+<ext:Extension xmlns:ext="__ExtensionPoint__"
+        xmlns:what="123.456.789">
+        <what:ItIsNotKnown>
+        dsafasdf
+        <what:Unknown>what:lorem</what:Unknown>
+        </what:ItIsNotKnown>
+</ext:Extension>""") # noqa: S320
+        inst1 = ExtensionLocalValue([xml1])
+        inst2 = ExtensionLocalValue([xml2])
+        self.assertEqual(inst1, inst2)
```
Comment (leon1995) [inline tests/test_pmtypes.py:263]: add empty line at the end

File: src/sdc11073/mdib/containerproperties.py [hunk 7]
```diff
@@ -456,45 +456,103 @@ def updateXMLValue(self, instance, node):
...
+
+    return all(map(_compare_extension, left_children, right_children))  # compare children but keep order
+
+
+class ExtensionLocalValue(list):
+
+    compare_method = _compare_extension
+    """may be overwritten by user if a custom comparison behaviour is required"""
+
+    # def __init__(self, value):
```
Comment (leon1995) [inline src/sdc11073/mdib/containerproperties.py:487]: needed?

**Review by leon1995 — CHANGES_REQUESTED** [review]

File: src/sdc11073/mdib/containerproperties.py (hunk 1 above)
Comment (deichmab-draeger) [reply src/sdc11073/mdib/containerproperties.py]: done

File: tests/test_pmtypes.py (hunk 5 above)
Comment (deichmab-draeger) [reply tests/test_pmtypes.py:71]: done

File: tests/test_pmtypes.py (hunk 6 above)
Comment (deichmab-draeger) [reply tests/test_pmtypes.py]: done

File: src/sdc11073/mdib/containerproperties.py (hunk 7 above)
Comment (deichmab-draeger) [reply src/sdc11073/mdib/containerproperties.py]: done

**Review by leon1995 — APPROVED** [review]

**Pull request merged**